openquake/hazardlib/gsim/megawati_2003.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2014-2017 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`megawatipan2010`.
+"""
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from scipy.constants import g
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+class MegawatiEtAl2003(GMPE):
+    """
+    Implements GMPE developed by Megawati, Pan and Koketsu and published in
+    2003 as "Response spectral attenuation relationships for Singapore and the
+    Malay Peninsula due to distant Sumatran-fault earthquakes", Earthquake
+    Engineering & Structural Dynamics Volume 32, pages 2241–2265.
+    """
+
+    #: Supported tectonic region type is active shallow crust
+    #: Sumatra strike-slip fault
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration,
+    #: peak ground veloacity and peak ground acceleration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is geometric mean
+    #: of two horizontal components,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation types is total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL
+    ])
+
+    #: No site parameter required. This GMPE is for very hard rock conditions
+    REQUIRES_SITES_PARAMETERS = set(())
+
+    #: Required rupture parameter is magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: Required distance measure is hypocentral distance, and azimuth
+    REQUIRES_DISTANCES = set(('rhypo', 'azimuth'))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # Check that the GSIM supports the standard deviations requested
+        assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
```

Comment:
This is also checked in https://github.com/gem/oq-hazardlib/blob/master/openquake/hazardlib/gsim/base.py#L516 in the `get_poes` method. However this method is called in very specific cases. Hence I would prefer to keep this check.